CMakeLists.txt

```diff
@@ -1,10 +1,275 @@
+        /wd4245  # Signed/unsigned mismatch
+        /wd4267  # Conversion from size_t
+        /wd4305  # Truncation
+        /wd4310  # Cast truncates constant
+        /wd4311  # Pointer truncation
+        /wd4324  # Structure padding
+        /wd4458  # Declaration hides member
+        /wd4464  # Relative include path
+        /wd4505  # Unreferenced function removed
+        /wd4514  # Unreferenced inline removed
+        /wd4530  # Exception handler
+        /wd4611  # setjmp interaction
+        /wd4702  # Unreachable code
+        /wd4710  # Function not inlined
+        /wd4711  # Selected for inline
+        /wd4820  # Padding added
+        /wd4996  # Deprecated functions
+        /wd5026  # Move ctor deleted
+        /wd5027  # Move assign deleted
+        /wd5045  # Spectre mitigation
+    )
+
+    target_compile_definitions(${target} PRIVATE
+        WIN32_LEAN_AND_MEAN
+        NOMINMAX
+        _CRT_SECURE_NO_WARNINGS
+        _CRT_NONSTDC_NO_WARNINGS
+        _SCL_SECURE_NO_WARNINGS
+        _HAS_EXCEPTIONS=1
+    )
+
+    set_target_properties(${target} PROPERTIES
+        MSVC_RUNTIME_LIBRARY "MultiThreaded$<$<CONFIG:Debug>:Debug>DLL"
+        VS_GLOBAL_CLToolChain "v143"
+        VS_GLOBAL_PlatformToolset "v143"
+        VS_GLOBAL_DisableSpecificWarnings "MSB8027;MSB8065;4244;4267;4996;4273;4141;4005;4141" 
+        VS_GLOBAL_TreatWarningAsError "false"
+        VS_GLOBAL_WarningLevel "0"
+    )
+
+    configure_cpu_optimized_unity_build(${target})
+endfunction()
+
+# Linux detection function
+function(configure_linux_target target)
+    if(WIN32)
+        return()
+    endif()
+
+    target_compile_options(${target} PRIVATE
+        -Wall
+        -Wextra
+        -Wpedantic
+        -Wno-unknown-pragmas
+        -fPIC
+        $<$<OR:$<CONFIG:Release>,$<CONFIG:Distribution>>:-O3>
+    )
+
+    if(WARNINGS_AS_ERRORS)
+        target_compile_options(${target} PRIVATE -Werror)
+    endif()
+
+    set_target_properties(${target} PROPERTIES
+        POSITION_INDEPENDENT_CODE ON
+        CXX_VISIBILITY_PRESET hidden
+        VISIBILITY_INLINES_HIDDEN ON
+    )
+endfunction()
+
+# Common target configuration function
+function(configure_target_build target)
+    if(WIN32 AND MSVC)
+        configure_windows_target(${target})
+        
+        target_include_directories(${target} BEFORE PRIVATE 
+            ${CMAKE_CXX_IMPLICIT_INCLUDE_DIRECTORIES}
+            ${PROJECT_BINARY_DIR}
+            ${PROJECT_SOURCE_DIR}/third_party/JoltPhysics
+        )
+    else()
+        configure_linux_target(${target})
+    endif()
+
+    target_compile_definitions(${target} PRIVATE
+        JPH_PROFILE_ENABLED=0
+        JPH_DEBUG_RENDERER=0
+        JPH_FLOATING_POINT_EXCEPTIONS_ENABLED=0
+        JPH_USE_SSE4_1=1
+        JPH_USE_SSE4_2=1
+        JPH_TRACK_BROADPHASE_STATS=0
+        JPH_STATISTICS=0
+        JPH_DISABLE_CUSTOM_ALLOCATOR
+    )
+endfunction()
+
+# Source files print function
+function(print_source_files target)
```

Comment:
Isn't this like debugging code? I'd remove this.

Reply:
this helps printing the files that are being compiled in the unity instead of just having the unity being compiled it is really important for the compile imho

Reply:
Long helper code like this should be moved to a separate file (probably fine to have in the scripts folder like `Scripts/CMakeHelperMethods.cmake`) for cleaning things up in the main files.